cli/src/test/java/com/devonfw/tools/ide/tool/intellij/IntellijJsonUrlUpdaterTest.java

```diff
@@ -7,48 +7,74 @@
 import static com.github.tomakehurst.wiremock.client.WireMock.urlMatching;
 
 import java.io.IOException;
+import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Path;
 
 import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.io.TempDir;
 
 import com.devonfw.tools.ide.url.model.folder.UrlRepository;
 import com.devonfw.tools.ide.url.updater.JsonUrlUpdater;
+import com.github.tomakehurst.wiremock.junit5.WireMockRuntimeInfo;
 import com.github.tomakehurst.wiremock.junit5.WireMockTest;
 
 /**
  * Test class for integrations of the {@link IntellijUrlUpdater}
  */
-@WireMockTest(httpPort = 8080)
+@WireMockTest
 public class IntellijJsonUrlUpdaterTest extends Assertions {
 
-  /**
-   * Test resource location
-   */
+  /** Test resource location */
   private final static String TEST_DATA_ROOT = "src/test/resources/integrationtest/IntellijJsonUrlUpdater";
 
   /** This is the SHA256 checksum of aBody (a placeholder body which gets returned by WireMock) */
   private static final String EXPECTED_ABODY_CHECKSUM = "de08da1685e537e887fbbe1eb3278fed38aff9da5d112d96115150e8771a0f30";
 
+  /** Temporary directory for the version json file */
+  @TempDir
+  static Path tempVersionFilePath;
+
+  /**
+   * Creates an intellij-version and intellij-version-without-checksum json file based on the given test resource in a temporary directory according to the http
+   * url and port of the {@link WireMockRuntimeInfo}.
+   *
+   * @param wmRuntimeInfo wireMock server on a random port
+   * @throws IOException
+   */
+  @BeforeAll
+  public static void setupTestVersionFile(WireMockRuntimeInfo wmRuntimeInfo) throws IOException {
+    //preparing test data with dynamic port
+    assertThat(Files.isDirectory(tempVersionFilePath)).isTrue();
+    String content = new String(Files.readAllBytes(Path.of(TEST_DATA_ROOT).resolve("intellij-version.json")), StandardCharsets.UTF_8);
+    content = content.replaceAll("\\$\\{testbaseurl\\}", wmRuntimeInfo.getHttpBaseUrl());
+    Files.write(tempVersionFilePath.resolve("intellij-version.json"), content.getBytes(StandardCharsets.UTF_8));
+
+    content = new String(Files.readAllBytes(Path.of(TEST_DATA_ROOT).resolve("intellij-version-without-checksum.json")), StandardCharsets.UTF_8);
+    content = content.replaceAll("\\$\\{testbaseurl\\}", wmRuntimeInfo.getHttpBaseUrl());
+    Files.write(tempVersionFilePath.resolve("intellij-version-without-checksum.json"), content.getBytes(StandardCharsets.UTF_8));
```

Comment:
Here we also write dynamic content to files.
Great thing here is that you are using `@TempDir` so it is fine.
However, writing a file in `@BeforeAll` to disc just for the sake to later read it in the test from the disk again seems also kind of pointless.
Instead of using the `@TempDir` you could simply store the content as String in a variable and take it from there instead of using the disc for communication.
There are some other places with the same pattern.
Surely a nice-to-have improvement but would still make sense to me to improve it...